src/ast.rs

```diff
@@ -1075,34 +533,11 @@ impl CheckParse for AndorJob {
     }
 }
 
-define_list_node!(AndorJobList, AndorJob);
-
-/// A freestanding_argument_list is equivalent to a normal argument list, except it may contain
-/// TOK_END (newlines, and even semicolons, for historical reasons).
-/// In practice the tok_ends are ignored by fish code so we do not bother to store them.
```

Comment:
it's weird that there was no difference to ArgumentList,
because ArgumentList definitely should not contain newlines or semicolons.
AFAICT that difference is not touched by your changes, so that's fine.

But we should definitely consider preserving the comment describing this historical weirdness.
Can you move the comment to `fn parse_argument_list` ?